weasyl/controllers/user.py

```diff
@@ -42,6 +46,32 @@ def signin_post_(request):
         # Invalidate cached versions of the frontpage to respect the possibly changed SFW settings.
         index.template_fields.invalidate(logid)
         raise HTTPSeeOther(location=form.referer)
+    elif logid and logerror == "2fa":
+        # Password authentication passed, but user has 2FA set, so verify second factor (Also set SFW mode now)
+        if form.sfwmode == "sfw":
+            request.set_cookie_on_response("sfwmode", "sfw", 31536000)
+        index.template_fields.invalidate(logid)
+        # Check if out of recovery codes; this should *never* execute normally, save for crafted
+        #   webtests. However, check for it and log an error to Sentry if it happens.
+        remaining_recovery_codes = two_factor_auth.get_number_of_recovery_codes(logid)
+        if remaining_recovery_codes == 0:
+            raise RuntimeError("Two-factor Authentication: Count of recovery codes for userid " +
+                               str(logid) + " was zero upon password authentication succeeding, " +
+                               "which should be impossible.")
```

Comment:
Charmander is correct, here. Using [this](https://testing-old.weasyl.com/search?within=all&cat=1000&q=test&nextid=test&find=submit) as an example:

```
An unexpected error occurred and your request could not be fully processed.
If you wish to contact support (on the forums or by e-mail) about this error, please include the following information: 
request-id: 180b7a29227440a3933105fe2d634986-2lCZ1kxv
```

Where **on the forums** and **by e-mail** are hyperlinked to the location you would expect them to go to (site/forum support and support@...).